src/components/searchablePoolList/SearchablePoolList.tsx

```diff
@@ -0,0 +1,136 @@
+import { useEffect, useState } from 'react';
+import { InputField } from 'components/inputField/InputField';
+import { useAppSelector } from 'redux/index';
+import { Token } from 'services/observables/tokens';
+import { Modal } from 'components/modal/Modal';
+import { Image } from 'components/image/Image';
+import { Pool } from 'services/api/bancor';
+import { createListPool, ListPool, wait } from 'utils/pureFunctions';
+
+interface SearchablePoolListProps {
+  onClick: (pool: ListPool) => void;
+  isOpen: boolean;
+  setIsOpen: Function;
+  excludedPoolAnchors?: string[];
+  includedPoolAnchors?: string[];
+}
+
+export const SearchablePoolList = ({
+  onClick,
+  isOpen,
+  setIsOpen,
+  excludedPoolAnchors = [],
+  includedPoolAnchors = [],
+}: SearchablePoolListProps) => {
+  const [search, setSearch] = useState('');
+
+  const pools = useAppSelector<Pool[]>((state) => state.bancor.pools);
+  const tokens = useAppSelector<Token[]>((state) => state.bancor.tokens);
+
+  const listPools = pools
```

Comment:
this filtering and mapping of the pools array happens every time this component is rendered. This is a derived state and it should be a selector (redux).